### PR TITLE
test: remove useless ASSERT_XXX macros for rgw test

### PR DESCRIPTION
**Pull request by david-z**
Signed-off-by: Zhi Zhang <zhangz.david@outlook.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`

</details>

### CODE REVIEW

**Comment by david-z** [comment]
retest this please

File: src/test/rgw/test_rgw_manifest.cc [hunk 1]
```diff
@@ -25,8 +25,8 @@
 #define ASSERT_EQ(v, s) if(v != s)cout << "Error at " << __LINE__ << "(" << #v << "!= " << #s << "\n"; \
                                 else cout << "(" << #v << "==" << #s << ") PASSED\n";
 #define EXPECT_EQ(v, s) ASSERT_EQ(v, s)
-#define ASSERT_TRUE(c) if(c)cout << "Error at " << __LINE__ << "(" << #c << ")" << "\n"; \
-                          else cout << "(" << #c << ") PASSED\n";#define EXPECT_TRUE(c) ASSERT_TRUE(c) 
+#define ASSERT_TRUE(c) if(!c)cout << "Error at " << __LINE__ << "(" << #c << ")" << "\n"; \
```
Comment (tchaikov) [inline src/test/rgw/test_rgw_manifest.cc]: @david-z under what circumstances is this chunk compiled? as i found that we have
```c++
#define GTEST
#ifdef GTEST
...
#else
...
```

Comment (david-z) [reply src/test/rgw/test_rgw_manifest.cc]: @tchaikov  yes, we used gtest, so this error was never came out. I just noticed this code error by accident, so submit the fix.

Comment (tchaikov) [reply src/test/rgw/test_rgw_manifest.cc]: @david-z thanks. the change per-se looks good apart from some formatting nits..

but, actually, i am in favor of removing them as a whole. these GTEST macro replacements were introduced by #1214.

@yehudasa @jdurgin do you recall why we have them? can we simply ditch these macros?

Comment (jdurgin) [reply src/test/rgw/test_rgw_manifest.cc]: @tchaikov I've no idea where they came from

Comment (david-z) [reply src/test/rgw/test_rgw_manifest.cc]: @tchaikov  do we ditch these macros anyway? I think it is totally fine.

Comment (tchaikov) [reply src/test/rgw/test_rgw_manifest.cc]: @david-z agreed, i am all for removing dead code! could you repurpose this PR to do so?

Comment (david-z) [reply src/test/rgw/test_rgw_manifest.cc]: Sure, will do.

File: src/test/rgw/test_rgw_manifest.cc [hunk 2]
```diff
@@ -20,14 +20,6 @@
 #define GTEST
 #ifdef GTEST
 #include <gtest/gtest.h>
-#else
```
Comment (tchaikov) [inline src/test/rgw/test_rgw_manifest.cc]: i'd suggest remove
```c++
#define GTEST
#ifdef GTEST
```
and the closing
```c++
#endif
```
as well.

Comment (david-z) [reply src/test/rgw/test_rgw_manifest.cc]: done

**Comment by tchaikov** [comment]
jenkins test signed

**Comment by tchaikov** [comment]
jenkins test submodules

**Comment by tchaikov** [comment]
jenkins test signed

**Comment by tchaikov** [comment]
jenkins test signed

**Comment by tchaikov** [comment]
@cbodley Hi Casey, i looked around but failed to find the test / test suite exercising any of `ceph_test_rgw_manifest`, `ceph_test_rgw_obj`, `ceph_test_cls_rgw_log` and `test_cors`.

also i found that some of them were disabled by @liewegas in a44854cd5539595c2a178200dd3ff8552234fb09.

**Pull request merged**